packages/atomic/src/components/facets-v1/atomic-facet/atomic-facet.tsx

```diff
@@ -0,0 +1,163 @@
+import {Component, h, State, Prop} from '@stencil/core';
+import {
+  Facet,
+  buildFacet,
+  FacetState,
+  FacetOptions,
+  FacetSortCriterion,
+  SearchStatus,
+  SearchStatusState,
+  buildSearchStatus,
+  FacetValue,
+} from '@coveo/headless';
+import {
+  Bindings,
+  BindStateToController,
+  BindStateToI18n,
+  I18nState,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../utils/initialization-utils';
+import {FacetPlaceholder} from '../../facets/atomic-facet-placeholder/atomic-facet-placeholder';
+import {FacetWrapper} from '../facet-wrapper/facet-wrapper';
+import {FacetHeader} from '../facet-header/facet-header';
+import {FacetSearchInput} from '../facet-search-input/facet-search-input';
+import {FacetValueCheckbox} from '../facet-value-checkbox/facet-value-checkbox';
+import {FacetValueLink} from '../facet-value-link/facet-value-link';
+import {FacetValueBox} from '../facet-value-box/facet-value-box';
+import {FacetShowLess} from '../facet-show-less/facet-show-less';
+import {FacetShowMore} from '../facet-show-more/facet-show-more';
+
+/**
+ * A facet is a list of values for a certain field occurring in the results, ordered using a configurable criteria (e.g., number of occurrences).
+ * An `atomic-facet` displays a facet of the results for the current query. In mobile browsers, this is rendered as a button that opens a facet modal.
+ */
+@Component({
+  tag: 'atomic-facet-v1', // TODO: remove v1 when old facets are removed
+  styleUrl: 'atomic-facet.pcss',
+  shadow: true,
+})
+export class AtomicFacet implements InitializableComponent {
+  @InitializeBindings() public bindings!: Bindings;
+  public facet!: Facet;
+  public searchStatus!: SearchStatus;
+
+  @BindStateToController('facet', {subscribeOnConnectedCallback: true})
+  @State()
+  public facetState!: FacetState;
+  @BindStateToController('searchStatus')
+  @State()
+  private searchStatusState!: SearchStatusState;
+  @State() public error!: Error;
+
+  /**
+   * Specifies a unique identifier for the facet.
+   */
+  @Prop({mutable: true, reflect: true}) public facetId?: string;
+  /**
+   * The non-localized label for the facet.
+   */
+  @Prop() public label = 'noLabel';
+  /**
+   * The field whose values you want to display in the facet.
+   */
+  @Prop() public field!: string;
+  /**
+   * The number of values to request for this facet.
+   * Also determines the number of additional values to request each time more values are shown.
+   */
+  @Prop() public numberOfValues = 8;
+  /**
+   * Whether this facet should contain a search box.
+   */
+  @Prop() public withSearch = true;
```

Comment:
Maybe a better precision would be to say that this also depends on the number of values returned for a given facet ( ie: > numberOfValues )